### PR TITLE
fix: 'Unsaved Settings Warning' pops up for no reason #450

### LINKED ISSUE #450: 'Unsaved Settings Warning' pops up for no reason
### Description

When navigating to the **"Settings" tab**, there is a brief and inconsistent display of the **"Unsaved Settings" warning**. This warning appears for a **split second,** even when there are no unsaved changes. It should **only be displayed when there are actual unsaved changes** in the settings.

### Screenshots

_Current Display:_ 

https://github.com/Dun-sin/Whisper/assets/110025628/a77c024c-8c81-46ea-a9c4-1ce16a220477

_Ideal Display:_

https://github.com/Dun-sin/Whisper/assets/110025628/1347aa37-3b96-4d84-8b22-f3f91d9306f2

### Additional information

Its essential to ensure that the warning displays **only when the user makes appropriate changes**.

### 🥦 Browser

Google Chrome

### 👀 Have you checked if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read and understood the rules in the [Contributing Guidelines](https://github.com/Dun-sin/Whisper/blob/main/CONTRIBUTING.md)

### Are you willing to work on this issue ?

Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by myatthiha-ucsy26**
# Fixes Issue

**My PR closes #450 **

# 👨‍💻 Changes proposed(What did you do ?)

I modified the code to ensure that warning message is displayed only when there are actual unsaved changes by checking 
 `hasUnsavedSettings`. 

# ✔️ Check List (Check all the applicable boxes)

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.

##  Note to reviewers

# 📷 Screenshots

<details> <summary> After Fix Evidence </summary>

https://github.com/Dun-sin/Whisper/assets/98248371/bfc8fb40-9630-4dd9-ae07-065c1be4c5f3

</details>

<details> <summary>Code Changes</summary> 

image

</details>

### CODE REVIEW

**Comment by myatthiha-ucsy26** [comment]
@Dun-sin Please kindly review it. And there are some conflict. Do I need to solve it ? You can check my code changes in PR description.

File: client/src/pages/Settings.jsx [hunk 1]
```diff
@@ -176,7 +176,8 @@ const Searching = () => {
                                 />
                                 <p className="dark:text-highlight text-red text-lg">Warning: You have unsaved settings</p>
                             </div>
-                        </Animation.Bounce>
+                        </Animation.Bounce>}
+
```
Comment (Dun-sin) [inline client/src/pages/Settings.jsx:180]:
```suggestion
```

**Comment by Dun-sin** [comment]
delete this PR, take note of the change i did, remove all changes you did, delete your fork, fork it again with the latest code apply your issue change and make the PR again.

**Comment by myatthiha-ucsy26** [comment]
OK. I will do it again.

**Pull request closed without merging**